proposals/0239-operator-designated-types.md

```diff
@@ -0,0 +1,137 @@
+# Extending operator declarations with designated types
+
+* Proposal: [SE-0239](0239-operator-designated-types.md)
+* Authors: [Mark Lacey](https://github.com/rudkx)
+* Review Manager: TBD
+* Status: **Implemented under staging options**
+
+*During the review process, add the following fields as needed:*
+
+* Implementation: This is currently in-tree, disabled by default, and enabled by staging options (`-enable-operator-designated-types`, `-solver-enable-operator-designated-types`).
+* Decision Notes: [Rationale](https://forums.swift.org/), [Additional Commentary](https://forums.swift.org/)
+* Bugs: [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN), [SR-MMMM](https://bugs.swift.org/browse/SR-MMMM)
```

Comment:
```suggestion
* Bug: [SR-7847](https://bugs.swift.org/browse/SR-7847)
```

Reply:
@benrimmington Did you confirm that this is fixed with these changes enabled?

Reply:
Sorry, I didn't check. Maybe @dan-zheng can help.

Reply:
Okay, there are a bunch of cases where this will help and I'd rather not list ones that we don't know are helped for certain. I might make a quick pass at listing some that I can validate still repro without this change and are fixed by it, if I can make the time.